Fix hireling_lib crash on server startup

jlcvp opened the pull request:
# Description
- Using FLAG_NOLIMIT when returning the hireling lamp to user's store inbox.
- Ignoring the inbox available slots in this case avoids a crash in the hireling_lib file when the user losing the house has a full store inbox.


## Behaviour
### **Actual**
hireling_lib.lua crashes on a specific scenario on server startup

### **Expected**

Server startup must not crash
## Fixes
Fixes opentibiabr/canary#607 

## Type of change

Please delete options that are not relevant.

  - [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested

  - [x] Using a player without any available slots in the store inbox, that owns a house with a hireling spawned inside, setup this house to be freed upon the server save. On the server save, the hireling lamp is added to the store inbox and the startup continues without errors.

**Test Configuration**:

  - Server Version: 
  - Client:
  - Operating System:

## Checklist

  - [ ] My code follows the style guidelines of this project
  - [ ] I have performed a self-review of my own code
  - [ ] I checked the PR checks reports
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [ ] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

gpedro commented:
LGTM. Should check the version before sending, no?

Lycanzito commented:
it should be colored, right? '-' 
![image](https://user-images.githubusercontent.com/52710697/72118231-35f5c480-332f-11ea-8b09-f2f6741aaaa7.png)

mattyx14 commented:
he say it: missing creaturescript at scripts/creaturescripts yet

lBaah commented:
This is meant to use with OTBR not custom servers @Lycanzito .

@gpedro we should not support outdated client. This allows us to focus on the development of protocol 12 features.
Cipsoft has changed the way few things are rendered in the screen (12.15+) thus making outdated not compatible anymore, if we update the protocol more than 12.00. So we drop the support.

lBaah commented:
I'll close this since it is going to be a 1.0 feature.

jlcvp committed:
Fix hireling_lib crash on server startup

- Using FLAG_NOLIMIT when returning the hireling lamp to user's store inbox.
- Ignoring the inbox available slots in this case avoids a crash in the hireling_lib file when the user losing the house has a full store inbox.
